src/main/scala/Utilities.scala

```diff
@@ -0,0 +1,40 @@
+import org.apache.log4j.Level
+import java.util.regex.Pattern
+import java.util.regex.Matcher
+
+object Utilities {
+  /** Makes sure only ERROR messages get logged to avoid log spam. */
```

Comment:
* that may limit troubleshooting capabilities on production, hence at lest INFO level should be used.
* logging configuration should be externalized into a resource file, so it can be tuned on dev / test / production environment without intervention into code.